src/material/button/button.scss

```diff
@@ -63,6 +63,11 @@
   }
 }
 
+.mdc-button {
+  line-height: 36px;
```

Comment:
From what I can tell, there is no token in the button for `line-height`. Same for `text-decoration`. This is what prompted my earlier question to Miles

Reply:
Oh that's weird, normally they have line height in all the tokens. That being said, I'm not sure that adding a `line-height` here is the right thing to do since the button doesn't have a `line-height` now either. Maybe the root cause is that the `container-height` token isn't set?

Reply:
Nope, the `container-height` isn't the root cause. This is a little more convoluted than I thought, but the line-height was getting set before by both the `without-ripple` in _button-theme.scss and the `static-styles-without-ripple` in _button.scss. I'm going to do a bit more debugging to understand what exactly is the situation with gmat but my current understanding is that the line-height was getting set before so we will need to set it somehow - also, I don't know what the proper way to do that atm

Reply:
I can see that previously `.mdc-button` had `var(--mdc-typography-button-line-height, 36px)`, but there's another `.mdc-button` style that is overriding it to `inherit`: 
<img width="323" alt="image" src="https://github.com/angular/components/assets/4450522/4e08d47f-efb7-459e-91a9-88b530df527b">

Checking the computed value shows that it's set to `normal` right now:
<img width="296" alt="image" src="https://github.com/angular/components/assets/4450522/d3ad96f1-c7b4-4f16-a1f5-f088573c2e57">

It seems like it's the same situation in the gmat button.

Reply:
Yeah this is what threw me off. For some reason on gmat it's the 36px that's winning. I'm not certain why yet. I'll do some more digging and get back to you

<img width="525" alt="Screenshot 2023-10-27 at 10 00 39 AM" src="https://github.com/angular/components/assets/25158423/f6d95093-8765-4c33-ab89-5acb38dead4c">

<img width="526" alt="Screenshot 2023-10-27 at 10 01 20 AM" src="https://github.com/angular/components/assets/25158423/059f5bdc-aa07-4776-aff4-8451c401b986">

Reply:
It's possible that the component styles are being included multiple times.

Reply:
I think it's the order that the styles get loaded in. Idk why it's loaded in a different order for gmat yet, so I'll need to look more into that

gmat:
<img width="525" alt="Screenshot 2023-10-27 at 10 29 05 AM" src="https://github.com/angular/components/assets/25158423/1ba33213-680c-48e2-98f8-6f6972598b93">

mat:
<img width="525" alt="Screenshot 2023-10-27 at 10 29 12 AM" src="https://github.com/angular/components/assets/25158423/e101ec98-cb06-484d-9138-85dea659de95">

Reply:
Moving the style change into `_button-theme.scss` instead of `_button.scss` fixes the specificity issue for both mat and gmat, but I'm still not happy with the solution. IMO it makes more sense to have both mat & gmat have `line-height: normal`

Reply:
Yeah agreed that we don't want to perpetuate this. If it's too hard to change them all to normal, we should keep the 36px to gmat.

Reply:
Quick side-note here: I ran a tgp for this and ran into hundreds of failures that are triggered by this because depending on how a project is loading its theme styles, their buttons may have line-height 36 or inherit

Once I have a response from mdc on this, I might choose to land a initial change to fix this inconsistency before coming back to land the swap to tokens